Makefile

```diff
@@ -593,7 +593,13 @@ scalapkg:
 			-Dcurrent_libdir="$(ROOTDIR)/lib" \
 			-Dlddeps="$(LIB_DEP) $(ROOTDIR)/lib/libmxnet.a")
 
-scalatest:
+scalaunittest:
+	(cd $(ROOTDIR)/scala-package; \
+		mvn integration-test -P$(SCALA_PKG_PROFILE),$(SCALA_VERSION_PROFILE) -Dcxx="$(CXX)" \
+			-Dcflags="$(CFLAGS)" -Dldflags="$(LDFLAGS)" \
+			-Dlddeps="$(LIB_DEP) $(ROOTDIR)/lib/libmxnet.a" $(SCALA_TEST_ARGS))
+
+scalaintegrationtest:
```

Comment:
where do you specify to run tests only under examples ? does it also run unit tests?

Reply:
Currently, all unit test are run on `integration-test` stage in Maven and all integration test runs in `verify` stage. 

Yes, it runs unit test as well

Reply:
I am not sure I see the benefit of creating another stage that runs twice. Am I missing anything ? consider adding the test phase in examples under a maven profile which can passed when you call make integrationtests ?

Reply:
It works like this way:
if you put test in the `test` phase, test will be running in `test`, `package` and all after phases.
If you put test in the `integration-test`, test will be running in `verify`, but not in `package`. 
So it is impossible to keep test runs on a individual phase.

Reply:
Some alternative is applicable for tests done in Java. However we are using `scalatest` plugin, the plugins for Java surefire cannot used in here.

Reply:
Can you not run `verify` of examples under a called profile `integration-test`

Reply:
Now there is no verify. Only Integration test.

Reply:
I tried to centralize the two profiles in the main pom file instead of changing these properties in every module. The test plugins are spread in all the modules, After cleaning them and adding two profiles in the main pom <tagExcludes> configuration. The tests are now segfaulting, not sure what is causing. I will stop this effort here and we can see later.